packages/arb-token-bridge-ui/src/util/deposits/fetchTokenDepositsFromEventLogs.ts

```diff
@@ -0,0 +1,116 @@
+import { EventArgs, EventFetcher } from '@arbitrum/sdk'
+import { L1ERC20Gateway__factory } from '@arbitrum/sdk/dist/lib/abi/factories/L1ERC20Gateway__factory'
+import { DepositInitiatedEvent } from '@arbitrum/sdk/dist/lib/abi/L1ERC20Gateway'
+import { BlockTag } from '@ethersproject/providers'
+
+import { getProviderForChainId } from '@/token-bridge-sdk/utils'
+import { dedupeEvents } from './helpers'
+
+/**
+ * Get the parent network events created by a deposit
+ * @param parentChainId
+ * @param gatewayAddress
+ * @param filter
+ * @param parentTokenAddress
+ * @param fromAddress
+ * @param toAddress
+ * @returns
+ */
+export async function getDepositEvents(
+  parentChainId: number,
+  gatewayAddress: string,
+  filter: { fromBlock: BlockTag; toBlock: BlockTag },
+  parentTokenAddress?: string,
+  fromAddress?: string,
+  toAddress?: string
+): Promise<(EventArgs<DepositInitiatedEvent> & { txHash: string })[]> {
+  const parentProvider = getProviderForChainId(parentChainId)
+
+  const eventFetcher = new EventFetcher(parentProvider)
+  const events = (
+    await eventFetcher.getEvents(
+      L1ERC20Gateway__factory,
+      contract =>
+        contract.filters.DepositInitiated(
+          null, // parentToken
```

Comment:
Can't we pass `parentTokenAddress` here?

Reply:
fairly sure it's a typechain typing issue
we are doing the same with withdrawals

<img width="868" alt="image" src="https://github.com/user-attachments/assets/96ba68b3-bec7-4b83-a9aa-a3e290c87ec0" />